### PR TITLE
#1589: add dedicated error message for missing companion for static members

### PR DESCRIPTION
**Pull request by esarbe**

### CODE REVIEW

File: compiler/test/dotty/tools/dotc/reporting/ErrorMessagesTests.scala [hunk 1]
```diff
@@ -1249,4 +1249,27 @@ class ErrorMessagesTests extends ErrorMessagesTest {
       val CyclicInheritance(symbol, _) :: Nil = messages
       assertEquals("class A", symbol.show)
     }
+
+  @Test def missingCompanionForStatic =
+    checkMessagesAfter("checkStatic") {
+      """
+        |object Foo {
+        |  @annotation.static def bar(): Unit = bar()
```
Comment (allanrenucci) [inline compiler/test/dotty/tools/dotc/reporting/ErrorMessagesTests.scala]: `def bar(): Unit = ()`

File: compiler/test/dotty/tools/dotc/reporting/ErrorMessagesTests.scala [hunk 2]
```diff
@@ -1249,4 +1249,27 @@ class ErrorMessagesTests extends ErrorMessagesTest {
...
+        |}
+      """.stripMargin
+    }.expect { (itcx, messages) =>
+      implicit val ctx: Context = itcx
+      val MissingCompanionForStatic(member) = messages.head
+      assertEquals(member.show, "method bar")
+    }
+
+  @Test def notMissingCompanionForStatic =
```
Comment (allanrenucci) [inline compiler/test/dotty/tools/dotc/reporting/ErrorMessagesTests.scala]: You can remote this test. Here we only test the content of error messages.
We already have tests for correct and incorrect programs: `tests/run/statics.scala`,  `tests/neg/static-no-companion.scala`

File: compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala [hunk 3]
```diff
@@ -1972,4 +1972,11 @@ object messages {
     val msg = hl"Cannot extend ${"sealed"} $pclazz in a different source file"
     val explanation = "A sealed class or trait can only be extended in the same file as its declaration"
   }
+
+  case class MissingCompanionForStatic(member: Symbol)(implicit ctx: Context) extends Message(MissingCompanionForStaticID) {
+    val msg = hl"${member.owner} does not have a companion class"
+    val kind = "Syntax"
+    val explanation =
+      hl"object that contains @static members must have a companion class."
```
Comment (allanrenucci) [inline compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala]:
```scala
hl"Object that contain ${"@static"} members must have a companion class."
```

Comment (esarbe) [reply compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: I propose a slight change:
`hl"An object that contains ${"@static"} members must have a companion class."`

Comment (allanrenucci) [reply compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: Sure

**Comment by allanrenucci** [comment]
Can you rebase?

**Comment by esarbe** [comment]
Of course.

**Comment by allanrenucci** [comment]
Thanks @esarbe!

**Pull request merged**